### PR TITLE
polish(gry) : polish reward model and td error

### PR DESCRIPTION
**Pull request by ruoyuGao**
## Description
- add the config table for each reward model
- remove unuse parameter
- add comment for td error

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [x] pass all the tests

### CODE REVIEW

File: ding/reward_model/gail_irl_model.py [hunk 1]
```diff
@@ -113,35 +112,50 @@ class GailRewardModel(BaseRewardModel):
...
         batch_size=64,
+        # (int) Size of the input: obs_dim + act_dim
         input_size=4,
+        # (int) Collect steps per iteration
         target_new_data_count=64,
+        # (int) Linear model hidden size
         hidden_size=128,
+        # (int) Expert dataset size
         collect_count=100000,
+        # (int) Clear buffer per fix iters
```
Comment (PaParaZz1) [inline ding/reward_model/gail_irl_model.py:158]: fixed

File: ding/reward_model/gail_irl_model.py [hunk 2]
```diff
@@ -113,35 +112,50 @@ class GailRewardModel(BaseRewardModel):
...
+        # Bigger "update_per_collect" means bigger off-policy.
+        # collect data -> update policy-> collect data -> ...
         update_per_collect=100,
+        # (int) How many samples in a training batch.
         batch_size=64,
+        # (int) Size of the input: obs_dim + act_dim
         input_size=4,
+        # (int) Collect steps per iteration
         target_new_data_count=64,
+        # (int) Linear model hidden size
```
Comment (PaParaZz1) [inline ding/reward_model/gail_irl_model.py:154]: why linear here, there maybe some more complicated networks

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/reward_model/drex_reward_model.py [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
         type='drex',
+        # (float) The step size of gradient descent.
         learning_rate=1e-5,
+        # (int) How many updates(iterations) to train after collector's one collection.
+        # Bigger "update_per_collect" means bigger off-policy.
+        # collect data -> update policy-> collect data -> ...
         update_per_collect=100,
+        # (int) How many samples in a training batch.
         batch_size=64,
         target_new_data_count=64,
```
[the remark this answers is not in the document]
Comment (ruoyuGao) [reply ding/reward_model/drex_reward_model.py:46]: I checked with author, it is a mistake, drex do not have target_new_data_count

File: ding/reward_model/gail_irl_model.py (hunk 2 above)
Comment (ruoyuGao) [reply ding/reward_model/gail_irl_model.py:154]: because our remodel network is linear, should we change this comment?

File: ding/reward_model/gail_irl_model.py [hunk 4]
```diff
@@ -113,35 +112,50 @@ class GailRewardModel(BaseRewardModel):
...
         learning_rate=1e-3,
+        # (int) How many updates(iterations) to train after collector's one collection.
+        # Bigger "update_per_collect" means bigger off-policy.
+        # collect data -> update policy-> collect data -> ...
         update_per_collect=100,
+        # (int) How many samples in a training batch.
         batch_size=64,
+        # (int) Size of the input: obs_dim + act_dim
         input_size=4,
+        # (int) Collect steps per iteration
```
[the remark this answers is not in the document]
Comment (ruoyuGao) [reply ding/reward_model/gail_irl_model.py:152]: added

File: ding/rl_utils/td.py [hunk 5]
```diff
@@ -50,6 +50,28 @@ def m_q_1step_td_error(
...
 ) -> torch.Tensor:
+    """
+    Overview:
+        Munchausen td_error for DQN algorithm, support 1 step td error.
+    Arguments:
+        - data (:obj:`m_q_1step_td_data`): the input data, m_q_1step_td_data to calculate loss
+        - gamma (:obj:`float`): discount factor
+        - tau (:obj:`float`): Entropy factor for Munchausen DQN
+        - alpha (:obj:`float`): Discount factor for Munchausen term
+        - criterion (:obj:`torch.nn.modules`): loss function criterion
```
[the remark this answers is not in the document]
Comment (ruoyuGao) [reply ding/rl_utils/td.py:61]: fixed

**Review by PaParaZz1 — APPROVED** [review]
